aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,709 @@
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+const (
+	image = "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_model", &resource.Sweeper{
+		Name: "aws_sagemaker_model",
+		F:    testSweepSagemakerModels,
+	})
+}
+
+func testSweepSagemakerModels(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+
+	req := &sagemaker.ListModelsInput{
+		NameContains: aws.String("terraform-testacc-sagemaker-model"),
+	}
+	resp, err := conn.ListModels(req)
+	if err != nil {
+		return fmt.Errorf("error listing models: %s", err)
+	}
+
+	if len(resp.Models) == 0 {
+		log.Print("[DEBUG] No sagemaker models to sweep")
+		return nil
+	}
+
+	for _, model := range resp.Models {
+		_, err := conn.DeleteModel(&sagemaker.DeleteModelInput{
+			ModelName: model.ModelName,
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"error deleting sagemaker model (%s): %s",
+				*model.ModelName, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSSagemakerModel_basic(t *testing.T) {
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig(rName, image),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "name",
+						fmt.Sprintf("terraform-testacc-sagemaker-model-%s", rName)),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "primary_container.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "primary_container.0.image", image),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "execution_role_arn"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "arn"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_model.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_tags(t *testing.T) {
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfigTags(rName, image),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.%", "1"),
```

Comment:
This check seems flakey, but only in CI. Passes fine locally. May be some sort of eventual consistency issue.

```
--- FAIL: TestAccAWSSagemakerModel_tags (17.66s)
    testing.go:538: Step 0 error: Check failed: Check 2/3 error: aws_sagemaker_model.foo: Attribute 'tags.%' expected "1", got "0"
```